### PR TITLE
Prevent requests to send as form urlencoded content type

### PR DESCRIPTION
**Pull request by maingoh**
When sending a binary content, it was sent through the `data` field. When no `files` parameter is provided requests defaults the content type to form-urlencoded which the API doesn't support anymore.
It is now sent through the files parameter as multipart.

### CODE REVIEW

File: deepomatic/http_helper.py [hunk 1]
```diff
@@ -114,46 +114,50 @@ def format_params(self, data):
                     data[key] = json.dumps(value)
         return data
 
-    def dump_json_for_multipart(self, data, files):
-        def recursive_json_dump(prefix, obj, data, files, omit_dot=False):
+    def dump_json_for_multipart(self, dic):
+        if dic is None:
```
Comment (vdel) [inline deepomatic/http_helper.py]: maybe use `data` instead of `dic` ? Or `data_dict` ?

File: deepomatic/http_helper.py [hunk 2]
```diff
@@ -114,46 +114,50 @@ def format_params(self, data):
...
                     data = json.dumps(data)
-                elif content_type.strip() == 'multipart/mixed':
-                    content_type = None  # will be automatically set to multipart
-                    data, files = self.dump_json_for_multipart(data, files)
-                else:
-                    raise DeepomaticException("Unsupported Content-Type")
+            elif content_type.strip() == 'multipart/mixed':
+                # If no files are provided, requests will default to form-urlencoded content type
+                # But the API doesn't support it.
+                if not files:
```
Comment (vdel) [inline deepomatic/http_helper.py]: So this means we cannot read frames from a camera and send them as binary ?

**Review by vdel — APPROVED** [review]

File: Jenkinsfile [hunk 3]
```diff
@@ -1,3 +0,0 @@
-node {
```
Comment (thomas-riccardi) [inline Jenkinsfile]: If you want to remove dmake CI builds, you should also remove the dmake.yml then (it is currently broken: it refers to `install_virtualenv.sh` which was removed in this PR).

File: deepomatic/http_helper.py (hunk 2 above)
Comment (maingoh) [reply deepomatic/http_helper.py]: As discussed it is still possible ;)

**Pull request merged**